app/routes/circulars._archive._index.tsx

```diff
@@ -239,10 +246,37 @@ export default function () {
   if (searchString) searchString = `?${searchString}`
 
   const [inputQuery, setInputQuery] = useState(query)
-  const clean = inputQuery === query
+  const [inputDateGte, setInputDateGte] = useState(startDate)
+  const [inputDateLte, setInputDateLte] = useState(endDate)
+  const [showContent, setShowContent] = useState(false)
+  const [showDateRange, setShowDateRange] = useState(false)
+  const clean =
+    inputQuery === query &&
+    inputDateGte === startDate &&
+    inputDateLte === endDate
 
   const submit = useSubmit()
 
+  const setFuzzyTime = (startDate?: string) => {
+    if (startDate === 'custom') {
+      setShowDateRange(true)
+    } else {
+      setShowDateRange(false)
+      setInputDateGte(startDate)
+    }
+    setInputDateLte('')
+  }
+
+  const setDateRange = () => {
+    setShowContent(false)
+    const params = new URLSearchParams(location.search)
+    if (inputDateGte) params.set('startDate', inputDateGte)
+    else params.delete('startDate')
+    if (inputDateLte) params.set('endDate', inputDateLte)
+    else params.delete('endDate')
+    window.location.search = params.toString()
+  }
```

Comment:
In React, we don't modify the location search params directly. We would employ the [`useSearchParams` hook](https://reactrouter.com/en/main/hooks/use-search-params).

However, you shouldn't need to do that either, because the user is submitting a form to perform the navigation, and _that_ updates the search params.

Reply:
ah, okay, I can revert it to the previous implementation then